CMakeLists.txt

```diff
@@ -4,6 +4,10 @@ if(POLICY CMP0051)
   cmake_policy(SET CMP0051 NEW)
 endif()
 
+if(POLICY CMP0054)
```

Comment:
After comparing the CMake in Cygwin and Linux, I realized that I misunderstood the behavior. I'll separate this part to another PR with other approach method.

In CMake 3.4.3, if `CMP0054` policy isn't set to any value, `cmake` uses the OLD behavior and issues a long warning message for every `if()` statement having a quoted variable whose value is an other variable's name. 
For example,
```
set(FOO 1)
set(VAR_A FOO)
set(VAR_B BAR)
if ("${VAR_A}" STREQUAL "FOO")
endif()
if ("${VAR_B}" STREQUAL "BAR")
endif()
```
issues `CMP0054` warning message for `"${VAR_A}"` and not for `"${VAR_B}"`.

Cygwin `cmake` defines the variable `CMAKE_SYSTEM_NAME` as `CYGWIN` and defines the variable `CYGWIN` as 1(TRUE).

Linux `cmake` defines the variable `CMAKE_SYSTEM_NAME` as `Linux` and there is no variable `Linux`.

So, if we didn't set any value to `CMP0054` policy, every `if ("${CMAKE_SYSTEM_NAME}" ...)` generated long warning message in Cygwin, but did not generate in Linux.